### PR TITLE
STY: Check pytest.raises is used context manager

### PR DESCRIPTION
**Pull request by gfyoung**
Follow-up to https://github.com/pandas-dev/pandas/pull/24297#issuecomment-447675032.

### CODE REVIEW

File: ci/code_checks.sh
Comment (gfyoung) [reply ci/code_checks.sh]: > Small things, but I'd place this validation next to the assert_raises_regex, as they are related. 

Fair enough.

> Also, I'd remove the comment and just add a short note to MSG, IMO that would be even clearer, and more compact.

The comment should stay because it explains the logic of the regex used to check.  That being said, I can expand the `MSG` string.

**Comment by gfyoung** [comment]
The check works as expected:

https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=5501

I will address comments from @datapythonista and comment out the check.

**Comment by jbrockmendel** [comment]
There’s still something like a thousand non-compliant usages right?

**Comment by jreback** [comment]
> There’s still something like a thousand non-compliant usages right?

@gfyoung right, don't we need to either comment this out or run it but not fail on it?

**Comment by gfyoung** [comment]
@jreback : As I mentioned [above](https://github.com/pandas-dev/pandas/pull/24332#issuecomment-448333161), I will comment out the check now that I have confirmed that it works.

**Comment by jreback** [comment]
@gfyoung sgtm.

File: ci/code_checks.sh
Comment (gfyoung) [reply ci/code_checks.sh]: Expanded the `MSG` and moved it next to `assert_raises_regex` as requested.  I also added a new `MSG` to print in the CI to remind us about enabling the check in the future.

**Review by datapythonista — APPROVED** [review]
lgtm

not sure what's your plan next, but my suggestion is to fix few cases yourself in a file or directory, activate those in this validation, and then create some issues labelled as "good first issue", so first time contributors have the opportunity to work with something as easy as this (and use your first PR as a reference, which will make things even easier for them).

Or if you really prefer to fix all them yourself, please do it in several PRs (I read that we've got some thousands of cases, I'm assuming that was said literally)

**Comment by jreback** [comment]
thanks @gfyoung 

yeah prob need a way to actually implement this. I think a few PR's might just do it

**Comment by simonjayhawkins** [comment]
@gfyoung : will probably need to ignore `pytest.raises` in commented out code..

`grep -r -e "[[:space:]] pytest.raises" pandas/tests|grep -rv -e "#.*pytest.raises" -`

would work.

also may need to ignore `pytest raises` in skipped tests. perhaps comment them out?

**Comment by gfyoung** [comment]
@simonjayhawkins : You're right that those are edge cases we would have to tackle.  However, as we're in no rush to invoke this linting yet, let's see how many of those we have (especially once `PY2` is phased out).  Commenting out is one option, but we could also just fill in with dummy messages for the time being (so that they can remind us to populate with meaningful strings once un-skipped).

**Comment by simonjayhawkins** [comment]
> but we could also just fill in with dummy messages for the time being

OK. that's simpler, i'll go with that approach if no objections.

> we're in no rush to invoke this linting yet

2 PRS awaiting approval, 1 PR should cover frame and another pytables, then moreless done.

**Comment by simonjayhawkins** [comment]
following #25516
```
$ grep -r -e "[[:space:]] pytest.raises" pandas/tests -l
```
now yields
```
pandas/tests/arrays/sparse/test_libsparse.py
pandas/tests/computation/test_eval.py
pandas/tests/frame/test_nonunique_indexes.py
pandas/tests/generic/test_generic.py
pandas/tests/generic/test_series.py
pandas/tests/indexes/interval/test_interval_new.py
pandas/tests/indexing/multiindex/test_partial.py
pandas/tests/indexing/test_floats.py
pandas/tests/io/test_pytables.py
pandas/tests/io/test_sql.py
pandas/tests/series/test_constructors.py
pandas/tests/test_panel.py
```

`pandas/tests/io/test_pytables.py` should probably wait until #25004 is merged.

**Pull request merged**
